### ISSUE TITLE
Integration with resumable.js and chunk deletion once buildFullFileFromChunks()

### ISSUE THREAD

**Issue by zedee** [pionl/laravel-chunk-upload#11]
Hi,
despite this library is compatible with the most popular uploaders, I've managed to integrate it in a project which uses dropzone.js as uploader and resumable.js for the chunk management. (I am unable to use an alternate upload system).

I've observed that chunks are not deleted once the `buildFullFileFromChunks()` is called in the `ChunkSave` class. In the case of uploading several times the same file, the last `.part` container grows and grows, as it has the same path.

My question is... why the chunks aren't deleted when the final file is merged? That'd also remove the need to run the `uploads:clear` command.

**Comment by pionl** [comment]
Hi @zedee, 

there is only single file when uploading and then it should be moved by you (when it's finished).

If you have left `.part` files, there is 2 reasons:
- the uploader is not sending cookie, then every upload makes a new chunk. You can change a config to not use session but IP address for naming
- something in the name of file changes

The clear command will delete old chunks (failed/cancelled uploads).

**Comment by zedee** [comment]
Thanks for your reply @pionl !
Okay, I think what's happening is that in my implementation, the saved file is passed to a `FileService` which then writes the stream to disk, hence the file is not moved but copied instead. (The thing is that I'm trying to implement the chunk upload with AsgardCMS).
So in `FileService` after the file is copied I'll have to delete the original chunk.
Is that a right approach?

**Comment by Chinese1904** [comment]
Why don't you just rename it?

```php
$chunk_file_path = "chunks/your-chunked-file.mp4.part;
$final_path = "public/uploads/your-chunked-file.mp4";
File::move($chunk_file_path,` $final_path);
```

`File::move` is a wrapper for PHP's `rename()` function. What I do is once the last chunk is uploaded, I just move it like I described.

**Comment by pionl** [comment]
@zedee @Chinese1904  Correct the desired function to use is the `move` method. Or if you want to upload to AWS:
1. Wait until last chunk is done
2. Upload the content to AWS
3. Delete the chunk file

**Comment by pionl** [comment]
Resolved in #12
